common-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java

```diff
@@ -74,47 +79,61 @@ public static String modifyMultipleMatchKeyword(String query){
                     List<Predicate> predicates = new ArrayList<>();
 
                     // Check that the concept name, code, or ID matches the query string.
-                    List<Predicate> queryPredicates = new ArrayList<>();
-                    Expression<Double> matchExp = criteriaBuilder.function("match", Double.class,
-                            root.get("conceptName"), criteriaBuilder.literal(keyword));
-                    queryPredicates.add(criteriaBuilder.greaterThan(matchExp, 0.0));
-                    queryPredicates.add(criteriaBuilder.equal(root.get("conceptCode"),
+                    List<Predicate> conceptCode_Id = new ArrayList<>();
+
+                    conceptCode_Id.add(criteriaBuilder.equal(root.get("conceptCode"),
                             criteriaBuilder.literal(query)));
+
                     try {
                         long conceptId = Long.parseLong(query);
-                        queryPredicates.add(criteriaBuilder.equal(root.get("conceptId"),
+                        conceptCode_Id.add(criteriaBuilder.equal(root.get("conceptId"),
                                 criteriaBuilder.literal(conceptId)));
                     } catch (NumberFormatException e) {
                         // Not a long, don't try to match it to a concept ID.
                     }
-                    predicates.add(criteriaBuilder.or(queryPredicates.toArray(new Predicate[0])));
+
+                    List<Predicate> conceptName_Filter = new ArrayList<>();
+
+
+                    Expression<Double> matchExp = criteriaBuilder.function("match", Double.class,
+                            root.get("conceptName"), criteriaBuilder.parameter(String.class , "keyword"));
+                    conceptName_Filter.add(criteriaBuilder.greaterThan(matchExp, 0.0));
+
 
                     // Optionally filter on standard concept, vocabulary ID, domain ID
                     if (standardConceptFilter.equals(StandardConceptFilter.STANDARD_CONCEPTS)) {
-                        predicates.add(criteriaBuilder.equal(root.get("standardConcept"),
+                        conceptName_Filter.add(criteriaBuilder.equal(root.get("standardConcept"),
```

Comment:
I'm wondering if we need to change this logic to include "C" here (and exclude it below). WDYT?

Reply:
Do you mean to include "C" concepts along with standard concepts and not return them as the part of non standard concepts ?

Reply:
Yeah. I'm not sure it makes sense for now... if I understand correctly, the "C" concepts are standard but because we haven't calculated descendant counts, they all have zero counts right now. So this is probably fine for the time being... eventually we may want to return C concepts with descendant counts.

Reply:
Added the check to include C concepts as part of standard concepts check and not include them as part of non standard concepts.